src/dash-playlist-loader.js

```diff
@@ -194,7 +194,7 @@ export const filterChangedSidxMappings = (masterXml, srcUrl, clientOffset, oldSi
 export const requestSidx_ = (loader, sidxRange, playlist, xhr, options, finishProcessingFn) => {
   const sidxInfo = {
     // resolve the segment URL relative to the playlist
-    uri: resolveManifestRedirect(options.handleManifestRedirects, sidxRange.resolvedUri),
+    uri: resolveManifestRedirect(options.handleManifestRedirects, sidxRange.resolvedUri, xhr),
```

Comment:
also, `xhr` in here isn't the request that `resolveManifestRedirect` expects but the actual method we use to make XHRs.